java/core/src/test/java/com/lancedb/lance/DatasetTest.java

```diff
@@ -296,6 +298,60 @@ void testAlterColumns() {
     }
   }
 
+  @Test
+  void testAddColumnBySqlExpressions() {
+    String testMethodName = new Object() {}.getClass().getEnclosingMethod().getName();
+    String datasetPath = tempDir.resolve(testMethodName).toString();
+    try (RootAllocator allocator = new RootAllocator(Long.MAX_VALUE)) {
+      TestUtils.SimpleTestDataset testDataset =
+          new TestUtils.SimpleTestDataset(allocator, datasetPath);
+      dataset = testDataset.createEmptyDataset();
+
+      SqlExpressions sqlExpressions = new SqlExpressions();
+      SqlExpressions.SqlExpression sqlExpression = new SqlExpressions.SqlExpression();
+      sqlExpression.setName("double_id");
+      sqlExpression.setExpression("id * 2");
+      sqlExpressions.setSqlExpressions(Collections.singletonList(sqlExpression));
+      dataset.addColumns(sqlExpressions, Optional.empty(), Optional.empty());
```

Comment:
Is this how java code is typically written? I'm not very familiar with Java but this seems quite verbose.

Unless it's really not idiomatic, I would consider a builder here too

```java
SqlExpressions sqlExpressions = SqlExpressions.Builder()
    .withExpression("double_id", "id * 2")
    .build();
dataset.addColumns(sqlExpressions, Optional.empty());
```

Reply:
That's OK. Generally, I would use Builder Pattern when I think there are many parameters and may add more in the future for a class.